middleware/middleware.go

```diff
@@ -70,6 +71,12 @@ func Error(name string, err error) error { return fmt.Errorf("%s/%s: %s", "middl
 // and a nil error.
 func NextOrFailure(name string, next Handler, ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
 	if next != nil {
+		span := ot.SpanFromContext(ctx)
+		if span != nil {
```

Comment:
Nit: if statement can be done on 1 line.
`if span := ot.SpanFromContext(ctx); span != nil {`